src/integration-builder/initialization.js

```diff
@@ -68,45 +68,67 @@ var initialization = {
         };
     },
     createConsentEvents: function () {
-        if (!window.Optanon) { return; }
-
-        var location = window.location.href,
-            consent,
-            consentState,
-            user = mParticle.Identity.getCurrentUser();
-
-        if (!user) {
+        var user = mParticle.Identity.getCurrentUser();
+        if (!window.Optanon || !user) {
             return;
         }
 
-        consentState = user.getConsentState();
+        var location = window.location.href;
+        var consentState =
+            user.getConsentState() || mParticle.Consent.createConsentState();
 
-        if (!consentState) {
-            consentState = mParticle.Consent.createConsentState();
-        }
         for (var key in this.consentMapping) {
             // removes all non-digits
             // 1st version of OneTrust required a selection from group1, group2, etc
             if (key.indexOf('group') >= 0) {
                 key = key.replace(/\D/g, '');
             }
-            var consentPurpose = this.consentMapping[key].purpose;
+
+            // Although consent purposes can be inputted into the UI in any casing
+            // the SDK will automatically lowercase them to prevent pseudo-duplicate
+            // consent purposes, so we call `toLowerCase` on the consentMapping purposes here
+            var consentPurpose = this.consentMapping[key].purpose.toLowerCase();
             var regulation = this.consentMapping[key].regulation;
-            var consentBoolean = false;
 
-            consentBoolean = (this.getConsentGroupIds().indexOf(key) > -1);
+            var newConsentBoolean = this.getConsentGroupIds().indexOf(key) > -1;
```

Comment:
`newConsentBoolean` does not really indicate what this is being used for. Why can't we continue calling it `consentBoolean`?

Reply:
I was trying to differentiate between the consent boolean on the current user.  I'll change this to `latestOTConsentBoolean`